editor/src/components/inspector/sections/component-section/component-section.tsx

```diff
@@ -372,6 +387,105 @@ function getSectionHeightFromPropControl(
   }
 }
 
+function usePropertyControlDescriptions(): Array<ControlDescription> {
+  return useGetPropertyControlsForSelectedComponents().flatMap((controls) =>
+    Object.values(controls.controls),
+  )
+}
+
+function arrayShapesMatch(left: Array<unknown>, right: Array<unknown>): boolean {
+  if (left.length === 0 || right.length === 0) {
+    return true
+  }
```

Comment:
this feels semantically wrong - we're asking if the array shapes match, but we are returning "yes!" if there's a difference (potential?) between them 🧐

Reply:
it may make sense functionally but then I'd either take it out of the function, or give it an explicit name